src/coreComponents/mesh/generators/VTKMeshGenerator.hpp

```diff
@@ -0,0 +1,268 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file VTKMeshGenerator.hpp
+ */
+
+#ifndef GEOSX_MESHUTILITIES_VTKMESHGENERATOR_HPP
+#define GEOSX_MESHUTILITIES_VTKMESHGENERATOR_HPP
+
+#include "dataRepository/Group.hpp"
+#include "codingUtilities/Utilities.hpp"
+#include "codingUtilities/StringUtilities.hpp"
+
+#include "MeshGeneratorBase.hpp"
+#include "mesh/CellBlockManager.hpp"
+
+#include <vtkSmartPointer.h>
+#include <vtkUnstructuredGrid.h>
+#ifdef GEOSX_USE_MPI
+#include <vtkMPIController.h>
+#include <vtkMPI.h>
+#else
+#include <vtkDummyController.h>
+#endif
+
+namespace geosx
+{
+
+
+/**
+ *  @class VTKMeshGenerator
+ *  @brief The VTKMeshGenerator class provides a class implementation of VTK generated meshes.
+ */
+class VTKMeshGenerator : public MeshGeneratorBase
+{
+public:
+/**
+ * @brief Main constructor for MeshGenerator base class.
+ * @param[in] name of the VTKMeshGenerator object
+ * @param[in] parent the parent Group pointer for the MeshGenerator object
+ */
+  VTKMeshGenerator( const string & name,
+                       Group * const parent );
+
+  ~VTKMeshGenerator() override = default;
+
+/**
+ * @brief Return the name of the VTKMeshGenerator in object Catalog.
+ * @return string that contains the key name to VTKMeshGenerator in the Catalog
+ */
+  static string catalogName() { return "VTKMeshGenerator"; }
+
+  /**
+   * @brief Create a new geometric object (box, plane, etc) as a child of this group.
+   * @param childKey the catalog key of the new geometric object to create
+   * @param childName the name of the new geometric object in the repository
+   * @return the group child
+   */
+  virtual Group * createChild( string const & childKey, string const & childName ) override;
```

Comment:
Also, just use `override`, no need for `virtual` (which is implied by `virtual` obviously).